[hl] -dce no breaks hl.I64

Using hl.I64 and `-dce no` together results in the following error:
```
[ERROR] /versions/4.3.1/std/hl/I64.hx:41: characters 12-38

 41 |  @:op(a+b) function add(v:I64) : I64;
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^
    | Missing function body
```

https://try.haxe.org/#8dad2c36

--- a/src/generators/genhl.ml
+++ b/src/generators/genhl.ml
@@ -3329,7 +3329,10 @@ let generate_static ctx c f =
 					let gen_content() =
 						op ctx (OThrow (make_string ctx ("Requires compiling with -D hl-ver=" ^ ver ^ ".0 or higher") null_pos));
 					in
-					ignore(make_fun ctx ~gen_content (s_type_path c.cl_path,f.cf_name) (alloc_fid ctx c f) (match f.cf_expr with Some { eexpr = TFunction f } -> f | _ -> abort "Missing function body" f.cf_pos) None None)
+					(match f.cf_expr with
+					| Some { eexpr = TFunction fn } -> ignore(make_fun ctx ~gen_content (s_type_path c.cl_path,f.cf_name) (alloc_fid ctx c f) fn None None)
+					| _ -> if not (Meta.has Meta.NoExpr f.cf_meta) then abort "Missing function body" f.cf_pos)
+					
 				else
 				add_native "std" f.cf_name
 			| (Meta.HlNative,[] ,_ ) :: _ ->
@@ -3337,7 +3340,9 @@ let generate_static ctx c f =
 			| (Meta.HlNative,_ ,p) :: _ ->
 				abort "Invalid @:hlNative decl" p
 			| [] ->
-				ignore(make_fun ctx (s_type_path c.cl_path,f.cf_name) (alloc_fid ctx c f) (match f.cf_expr with Some { eexpr = TFunction f } -> f | _ -> abort "Missing function body" f.cf_pos) None None)
+				(match f.cf_expr with
+				| Some { eexpr = TFunction fn } -> ignore(make_fun ctx (s_type_path c.cl_path,f.cf_name) (alloc_fid ctx c f) fn None None)
+				| _ -> if not (Meta.has Meta.NoExpr f.cf_meta) then abort "Missing function body" f.cf_pos)
 			| _ :: l ->
 				loop l
 		in

--- a/tests/misc/hl/projects/Issue11196/Issue11196.hx
+++ b/tests/misc/hl/projects/Issue11196/Issue11196.hx
@@ -1,0 +1,3 @@
+function main() {
+	var a:hl.I64 = 5;
+}

--- a/tests/misc/hl/projects/Issue11196/compile.hxml
+++ b/tests/misc/hl/projects/Issue11196/compile.hxml
@@ -1,0 +1,3 @@
+-m Issue11196
+-hl out.hl
+-dce no

